commit 44b09bb0434b572101feb75e0c1c186d98562135
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2019-08-13T00:15:29+00:00 (8 weeks ago)
    
    OS-7951 CUBIC module should react immediately to CC_RTO congestion signal

diff --git a/usr/src/uts/common/inet/cc/cc_cubic.c b/usr/src/uts/common/inet/cc/cc_cubic.c
index a4b8f29e18..57aee0f567 100644
--- a/usr/src/uts/common/inet/cc/cc_cubic.c
+++ b/usr/src/uts/common/inet/cc/cc_cubic.c
@@ -296,18 +296,13 @@ cubic_cong_signal(struct cc_var *ccv, uint32_t type)
 	case CC_RTO:
 		/*
 		 * Grab the current time and record it so we know when the
-		 * most recent congestion event was. Only record it when the
-		 * timeout has fired more than once, as there is a reasonable
-		 * chance the first one is a false alarm and may not indicate
-		 * congestion.
+		 * most recent congestion event was.
 		 */
-		if (CCV(ccv, tcp_timer_backoff) >= 2) {
-			cubic_data->num_cong_events++;
-			cubic_data->t_last_cong = gethrtime();
-			cubic_ssthresh_update(ccv);
-			cubic_data->max_cwnd = cwin;
-			CCV(ccv, tcp_cwnd) = mss;
-		}
+		cubic_data->num_cong_events++;
+		cubic_data->t_last_cong = gethrtime();
+		cubic_ssthresh_update(ccv);
+		cubic_data->max_cwnd = cwin;
+		CCV(ccv, tcp_cwnd) = mss;
 		break;
 	}
 }
